src/Microsoft.Health.SqlServer/ManagedIdentityAccessTokenHandler.cs

```diff
@@ -1,29 +1,27 @@
-﻿// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
 using EnsureThat;
-using Microsoft.Azure.Services.AppAuthentication;
 
 namespace Microsoft.Health.SqlServer;
 
 public class ManagedIdentityAccessTokenHandler : IAccessTokenHandler
 {
-    private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
-
-    public ManagedIdentityAccessTokenHandler(AzureServiceTokenProvider azureServiceTokenProvider)
+    /// <inheritdoc />
+    public async Task<string> GetAccessTokenAsync(string resource, CancellationToken cancellationToken)
     {
-        EnsureArg.IsNotNull(azureServiceTokenProvider, nameof(azureServiceTokenProvider));
+        EnsureArg.IsNotNullOrEmpty(resource, nameof(resource));
 
-        _azureServiceTokenProvider = azureServiceTokenProvider;
-    }
+        ManagedIdentityCredential credential = new ManagedIdentityCredential();
```

Comment:
Do we need a hook for passing the `clientId`? Or is that unnecessary?

Reply:
That is not necessary. It reads from environment variables.

Reply:
This code path gets executed only when one uses managed identity as their authentication. Also i verifed the authentication with both aad pod identity  and workload identity for our instances.

Reply:
But to be safe side updated to hook the clientid